kentik_api_library/kentik_api/synthetics/synth_tests/base.py

```diff
@@ -205,94 +221,46 @@ class HealthSettings(_ConfigElement):
-
-    def to_pb(self) -> pb.HealthSettings:
-        return pb.HealthSettings(
-            latency_critical=self.latency_critical,
-            latency_warning=self.latency_warning,
-            latency_critical_stddev=self.latency_critical_stddev,
-            latency_warning_stddev=self.latency_warning_stddev,
-            packet_loss_critical=self.packet_loss_critical,
-            packet_loss_warning=self.packet_loss_warning,
-            jitter_critical=self.jitter_critical,
-            jitter_warning=self.jitter_warning,
-            jitter_critical_stddev=self.jitter_critical_stddev,
-            jitter_warning_stddev=self.jitter_warning_stddev,
-            http_latency_critical=self.http_latency_critical,
-            http_latency_warning=self.http_latency_warning,
-            http_latency_critical_stddev=self.http_latency_critical_stddev,
-            http_latency_warning_stddev=self.http_latency_warning_stddev,
-            http_valid_codes=self.http_valid_codes,
-            dns_valid_codes=self.dns_valid_codes,
-            unhealthy_subtest_threshold=self.unhealthy_subtest_threshold,
-            activation=self.activation.to_pb(),
-        )
-
 
 @dataclass
 class SynTestSettings(_ConfigElement):
+    PB_TYPE = pb.TestSettings
+
     tasks: List[TaskType] = field(default_factory=list)
     family: IPFamily = Defaults.family
-    period: int = Defaults.period
+    period: int = Defaults.period  # in seconds
     agent_ids: List[ID] = field(default_factory=list)
     health_settings: HealthSettings = field(default_factory=HealthSettings)
     notification_channels: List[str] = field(default_factory=list)
 
     @classmethod
-    def task_name(cls) -> Optional[str]:
-        return None
-
-    def fill_from_pb(self, src: pb.TestSettings) -> None:
-        self.family = IPFamily(src.family)
-        self.period = src.period
-        self.agent_ids = [ID(id) for id in src.agent_ids]
-        self.tasks = [TaskType(task) for task in src.tasks]
-        self.health_settings.fill_from_pb(src.health_settings)
-        self.notification_channels = src.notification_channels
-
-    def to_pb(self) -> pb.TestSettings:
-        return pb.TestSettings(
-            family=self.family.value,
-            period=self.period,
-            agent_ids=[str(id) for id in self.agent_ids],
-            tasks=[task.value for task in self.tasks],
-            health_settings=self.health_settings.to_pb(),
-            notification_channels=self.notification_channels,
-        )
-
-    def to_dict(self) -> dict:
-        def _id(i: str) -> str:
-            try:
-                return f"{int(i):010}"
-            except ValueError:
-                return i
+    def from_pb(cls: Type[_ConfigElementT], obj: Any):
+        """Tweak inherited "from_pb" method to handle ignored and legacy task types"""
+
+        IGNORED_TASK_TYPES = ["bgp-monitor"]
+        LEGACY_TASK_TYPES = {"knock": "ping"}
+
+        obj = deepcopy(obj)
```

Comment:
Why does the object representing the protobuf need to be copied?

Reply:
So it doesn't modify the src object, this is against intuition that deserializing an object would modify it

Reply:
I have hard times to imagine any "mysterious error" that modification of the protobuf could cause in this case. The protobuf is discarded once the SDK object is constructed. The user (short of special hacks) has no way to access the protobuf that has been returned by the API. 
... but we can eventually change this later :).